python/ray/serve/schema.py

```diff
@@ -477,6 +477,35 @@ class ServeDeploySchema(BaseModel, extra=Extra.forbid):
         description=("The set of Serve applications to run on the Ray cluster."),
     )
 
+    @validator("applications")
+    def application_names_unique(cls, v):
+        # Ensure there are no duplicate applications listed
+        names = [app.name for app in v]
+        duplicates = {f'"{name}"' for name in names if names.count(name) > 1}
+        if len(duplicates):
+            apps_str = ("application " if len(duplicates) == 1 else "applications ") + (
+                ", ".join(duplicates)
+            )
+            raise ValueError(
+                f"Found multiple configs for {apps_str}. Please remove all duplicates."
+            )
+        return v
+
+    @validator("applications")
+    def application_routes_unique(cls, v):
+        # Ensure each application has a different route prefix
```

Comment:
Multiple apps may set `route_prefix` to `None` (so they're inaccessible over HTTP). Could you enable this behavior and add a test for it?